### ISSUE TITLE
busted doesn't work, busted_bootstrap does

### ISSUE THREAD

**Issue by ostinelli** [Olivine-Labs/busted#290]
Hello,
After installing the latest `2.0.rc3` I wanted to go back to the stable `1.11.1`. So I did, and this is what I have installed:

```
$ luarocks list

Installed rocks:
----------------
...
busted
   1.11.1-1 (installed) - /usr/local/lib/luarocks/rocks
...
```

When I run `busted` I get:

```
$ busted
luajit: /usr/local/bin/busted_bootstrap:3: unexpected symbol near '-'
```

When I run `busted_bootstrap` I get:
```
$ busted_bootstrap 

●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●● ○
195 successes / 0 failures / 0 pending : 0.141569 seconds.
```

The file `busted_bootstrap` contains:

```bash
#!/bin/sh

exec '/usr/local/opt/lua/bin/lua5.2' -e 'package.path="/Users/roberto/.luarocks/share/lua/5.2/?.lua;/Users/roberto/.luarocks/share/lua/5.2/?/init.lua;/usr/local/share/lua/5.2/?.lua;/usr/local/share/lua/5.2/?/init.lua;"..package.path; package.cpath="/Users/roberto/.luarocks/lib/lua/5.2/?.so;/usr/local/lib/lua/5.2/?.so;"..package.cpath' -e 'local k,l,_=pcall(require,"luarocks.loader") _=k and l.add_context("busted","1.11.1-1")' '/usr/local/lib/luarocks/rocks/busted/1.11.1-1/bin/busted_bootstrap' "$@"
```

My system:
```bash
$ lua -v
Lua 5.2.3  Copyright (C) 1994-2013 Lua.org, PUC-Rio
$ luajit -v
LuaJIT 2.0.3 -- Copyright (C) 2005-2014 Mike Pall. http://luajit.org/
```

Any ideas?

Thank you,
r.

**Comment by DorianGray** [comment]
Closing this since it only affects the rcs and changes between old busted and new.

**Comment by istr** [comment]
@lipp fix works for me too (linux; luarocks compiled from source). Thanks.

**Comment by blueyed** [comment]
@DorianGray 
What's the suggested workaround for this issue?

The problem appears to be that `/usr/local/bin/busted_bootstrap` is a shell script, and does appear to come from version 2, but gets installed for busted 1.11.1, too.

`/usr/local/lib/luarocks/rocks/busted/1.11.1-2/bin/busted_bootstrap` seems to be OK, but why isn't this file installed into `/usr/local/bin/busted_bootstrap`?

**Comment by blueyed** [comment]
Uninstalling busted removes the `/usr/local/bin/busted_bootstrap` file, but when installing `1.11.1`, the wrong version is getting installed there.

**Comment by blueyed** [comment]
And it does not appear to be related to having installed version 2 previously (see https://travis-ci.org/blueyed/awesome/builds/66254287).

**Comment by blueyed** [comment]
It seems to be related to `luarocks` creating a wrapper for/to `/usr/local/lib/luarocks/rocks/busted/1.11.1-2/bin/busted_bootstrap` after all.

**Comment by blueyed** [comment]
busted-1.11.1-2.rockspec has this, while version 2 only has the first one:

     install = {
        bin = {
          ["busted"] = "bin/busted",
          ["busted.bat"] = "bin/busted.bat",
          ["busted_bootstrap"] = "bin/busted_bootstrap"
        }
      }

**Comment by DorianGray** [comment]
Yes, for v 2.0 we changed how the test runner is loaded fairly drastically.

After reading this thread again, I think there are a few issues being conflated. Also, I cannot reproduce the issue on linux. I think tracking this one down may be up to you! If you do figure it out, please make a pull request so we can make a few more people happy.

**Comment by blueyed** [comment]
@DorianGray 
The linked build is a Linux machine (Travis).
I could imagine that some luarocks update / setting is involved here, which creates the wrapper around `busted_bootstrap`.  Maybe that could be disabled?

**Comment by istr** [comment]
I can replicate this issue with latest luarocks and busted-1.11.1 official rock reliably on linux. Even after completely fresh luarocks setup. Invariably @lipp s fix https://github.com/Olivine-Labs/busted/issues/290#issuecomment-62779977 resolves the issue after installing busted using the official luarock. Verified for lua 5.1 and lua 5.2 on Ubuntu 14.04.2 LTS machine with vanilla luarocks 2.2.2 built from sources.

**Comment by istr** [comment]
I think that I nailed down this problem:
the busted rockspec does not comply with what luarocks expects:
https://github.com/keplerproject/luarocks/wiki/Rockspec-format
see "build.install": ```build.install.bin (table) - Lua command-line scripts.```
bin/busted in 0e88edb9fb20c4446d194a15fa13bfbf9041c16f (1.11.1) is NOT a lua command line script, thus violates the rockspec format and this is why it fails.
Will make a PR for that soon. Just need to fix the rockspec (only include the lua script) and rename bin/busted_bootstrap to bin/busted for 1.11.1 I guess.

**Comment by blueyed** [comment]
@istr 
Makes sense.  Thanks!

**Comment by alexspetty** [comment]
To solve this problem on Ubuntu 15.04, I ran:
`luarocks remove busted --force`
`sudo luarocks install busted`
